### PR TITLE
Migrate frontend to serviceerror

### PR DESCRIPTION
**Pull request by alexshtin**

### CODE REVIEW

File: service/frontend/workflowHandler.go [hunk 1]
```diff
@@ -30,10 +30,10 @@ import (
 	"github.com/pborman/uuid"
 	commonproto "go.temporal.io/temporal-proto/common"
 	"go.temporal.io/temporal-proto/enums"
+	"go.temporal.io/temporal-proto/serviceerror"
 	"go.temporal.io/temporal-proto/workflowservice"
 	"google.golang.org/grpc/codes"
 
-	"github.com/temporalio/temporal/.gen/go/shared"
```
Comment (alexshtin) [inline service/frontend/workflowHandler.go]: Finally!

File: common/adapter/error.go [hunk 2]
```diff
@@ -173,3 +174,55 @@ func ToThriftError(st *status.Status) error {
 
 	return &shared.InternalServiceError{Message: fmt.Sprintf("temporal internal uncategorized error, msg: %s", st.Message())}
 }
+
+// ToProtoError converts Thrift error to gRPC error.
+func ToServiceError(in error) error {
```
Comment (alexshtin) [inline common/adapter/error.go]: Temporary mapper. Will be removed with all `adapter` package.

File: common/adapter/enums.go [hunk 3]
```diff
@@ -144,7 +144,7 @@ func ToThriftEncodingType(in enums.EncodingType) *shared.EncodingType {
 	case enums.EncodingTypeJSON:
 		ret := shared.EncodingTypeJSON
 		return &ret
-	case enums.EncodingTypeProto:
+	case enums.EncodingTypeProto3:
```
Comment (shawnhathaway) [inline common/adapter/enums.go]: Thanks!

File: common/adapter/error.go [hunk 4]
```diff
@@ -173,3 +174,55 @@ func ToThriftError(st *status.Status) error {
...
+	if in == nil {
+		return nil
+	}
+
+	switch thriftError := in.(type) {
+	case *shared.InternalServiceError:
+		return serviceerror.NewInternal(thriftError.Message)
+	case *shared.BadRequestError:
+		switch thriftError.Message {
+		case "No permission to do this operation.":
```
Comment (shawnhathaway) [inline common/adapter/error.go]: 😢

Comment (alexshtin) [reply common/adapter/error.go]: I will convert it to proper `PermissionDenied` error.

**Pull request merged**